projects/lang_feedback/base_gen.py

```diff
@@ -0,0 +1,119 @@
+# Unalignment Generation
+from __future__  import annotations
+import json
+import os
+import argparse
+
+import sys
+from tqdm.auto import tqdm
+from vllm import SamplingParams,LLM
+PROMPT_USER: str = 'USER: {prompt}'
+PROMPT_ASSISTANT: str = '\nASSISTANT:'  # should not have a space at the end
+PROMPT_INPUT: str = PROMPT_USER + PROMPT_ASSISTANT
+from datasets import load_dataset, concatenate_datasets
+from PIL import Image
+import requests
+
+import os
+
+# 设置CUDA可见设备
+os.environ["CUDA_VISIBLE_DEVICES"] = "0, 1, 2, 3, 4, 5, 6, 7" 
+
+def parse_arguments() -> argparse.Namespace:
+    """Parse the command-line arguments."""
+    parser = argparse.ArgumentParser(
+        description='Evaluate models with gpt4',
+    )
+    # Model
+    parser.add_argument(
+       '--model_name_or_path',
+        type=str,
+        help='the name or path of the first model (champion) in the arena to load from',
+    )
+    parser.add_argument(
+        '--input_path',
+        type=str,
+        help='the path of the input json file',
+    )
+    parser.add_argument(
+        '--output_name',
+        type=str,
+        help='the name of the output json file',
+        default=None,
+    )
+    parser.add_argument(
+        '--output_dir',
+        type=str,
+        help='Where to store the eval output.',
+    )
+    return parser.parse_args()
+
+
+def generate_answer_by_vllim(problems: list[str], model_name_or_path:str) ->list[str]:
+    samplingparams = SamplingParams(
+        temperature = 1.0,
+        repetition_penalty = 1.1,
+        max_tokens = 2048,
+        n=1,
+    )
+    
+    llm = LLM(model=model_name_or_path, 
+              gpu_memory_utilization=0.95, 
+              swap_space=32, 
+              trust_remote_code=True, 
+              tensor_parallel_size=8)  # 使用8个GPU并行
+    model_name = model_name_or_path.split('/')[-1]
+    outputs = llm.generate(problems, samplingparams)
+    answers = []
+    
+    for output, entry in tqdm(zip(outputs, problems)) :
+        items = []
+        for i in range(len(output.outputs)):
+            item = {
+                'from': model_name,
+                'response': output.outputs[i].text.strip()
+            }
+            items.append(item)
+        answers.append(items)
+    
+    return answers
+
+def main() -> None:
+    args = parse_arguments()
+    input_path = args.input_path
+    
+    with open(input_path, 'r', encoding='utf-8') as f:
+        data = json.load(f)
+    
+    print(f"input_path: {input_path}")
+ 
+    problems = []
+    for idx in range(len(data)):
+        image_file = data[idx]['image']
+        prompt = f"{data[idx]['prompt'].replace('<image>', ' image ').replace('<image>', ' image ')} <image>"
+        problem = {
+            'prompt': PROMPT_INPUT.format(prompt=prompt),
+            'multi_modal_data': {'image': Image.open(image_file)},
+        }
+        problems.append(problem)
+        
+    answers = generate_answer_by_vllim(problems, args.model_name_or_path)
```

Comment:
may be `vllm` no `vllim`, I should apologize for it, since this code may be some typos from previous codebase